### ISSUE TITLE
Autograder does not detect JSON score report if printed with a trailing newline

### ISSUE THREAD

**Issue by zackw** [autolab/Autolab#1528]
**Your environment**

- Carnegie Mellon's Autolab <https://autolab.andrew.cmu.edu/>
- Autograder running in rhelbeta.img

**Steps To Reproduce**

1. Write a lab with an autograding script that prints the JSON score report with a trailing newline afterward.
2. Make a submission for this lab.

**Current behavior**

The autograder will print an error message along the lines of "`syntax error, unexpected token ''` and the score report will not be processed.

**Expected behavior**

The score report should be processed.

**Comment by damianhxy** [comment]
Problem lies [here](https://github.com/autolab/Autolab/blob/47560c89ce32d10c2b52d139f29ae117a62753ce/app/helpers/assessment_autograde_core.rb#L348)

**Comment by zackw** [comment]
@damianhxy This is not to say that your proposed PR is wrong, but I don't understand how we got here in the first place, given that the `.lines` method (correctly) understands `\n` as a line _terminator_, not a line _separator_ --

```
irb(main):001:0> "abc\ndef\n".lines
=> ["abc\n", "def\n"]
```

**Comment by damianhxy** [comment]
@zackw By default, there will be a newline appended to the end of the Autograder output. Hence, when there is a trailing newline after the JSON score, there will actually be more than one trailing newline. As such, the additional newline(s) will be split into additional lines, causing the issue.

**Comment by zackw** [comment]
```python
        score_report = {
            "scores": scores_by_group,
            "scoreboard": [total_score]
        }
        json.dump(score_report, sys.stdout, sort_keys=True)
        sys.stdout.write("\n")
        sys.stdout.flush()
```

Python's `json.dump` does _not_ append a trailing newline.

> the issue arises when I have more than one trailing newline (i.e. `print("...\n")`)

Come to think of it, this code is being invoked from an `autograde-Makefile` that may be appending additional newlines.

**Comment by damianhxy** [comment]
I have tested your output code and verified that before the PR, autograding works both with and without the `sys.stdout.write("\n")`, but fails when I duplicate the write, so it is likely that you are appending additional newlines elsewhere.

I have also verified that with the PR, autograding works regardless of the number of newlines appended. If there are no further concerns, I'll proceed with merging the PR.
